Fix plugin related types for remarkable

pragdave opened the pull request:
If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <
See CHANGELOG.md.

Basically, the existing type definitions didn't extend to covering the plugin API

- [ ] Increase the version number in the header if appropriate.

n/a

- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

already there

andy-ms commented on package.json:
```diff
@@ -21,7 +21,10 @@
         "lint": "dtslint types"
     },
     "devDependencies": {
+        "ava": "^0.25.0",
```
I don't think these new dependencies are needed?

andy-ms commented on types/remarkable/CHANGELOG.md:
```diff
@@ -0,0 +1,10 @@
+2018-10-07 Sun  <pragdave@gmail.com>
```
We don't support CHANGELOG files -- these comments should go in the PR.

pragdave commented:
2018-10-07 Sun  <pragdave@gmail.com>

  * The type `Rule` was being used for both parsing and rendering rules,
    but it was really just the rendering side. Rather than rename it and
    break things, I added a new type for ParsingRule and updated the
    `Rulers` function signatures.

  * The `content` field in `ContentToken` was typed as a `string`.
    However, its type is really `any`, as it is used simply to transport
    values between the parser and the renderer for any given rule.


On Tue, Oct 9, 2018 at 4:04 PM Andy <notifications@github.com> wrote:

> *@andy-ms* commented on this pull request.
> ------------------------------
>
> In package.json
> <https://github.com/DefinitelyTyped/DefinitelyTyped/pull/29530#discussion_r223862406>
> :
>
> > @@ -21,7 +21,10 @@
>          "lint": "dtslint types"
>      },
>      "devDependencies": {
> +        "ava": "^0.25.0",
>
> I don't think these new dependencies are needed?
> ------------------------------
>
> In types/remarkable/CHANGELOG.md
> <https://github.com/DefinitelyTyped/DefinitelyTyped/pull/29530#discussion_r223862875>
> :
>
> > @@ -0,0 +1,10 @@
> +2018-10-07 Sun  <pragdave@gmail.com>
>
> We don't support CHANGELOG files -- these comments should go in the PR.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/DefinitelyTyped/DefinitelyTyped/pull/29530#pullrequestreview-163095567>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAApmNbN-pwKF1Lndv7oeOr_XHLeT8D2ks5ujQ9ngaJpZM4XMKqe>
> .
>

andy-ms commented on package.json:
```diff
@@ -22,6 +22,7 @@
     },
     "devDependencies": {
         "dtslint": "github:Microsoft/dtslint#production",
+        "ts-node": "^7.0.1",
```
This also does not seem to be needed.

andy-ms commented on types/remarkable/lib/index.d.ts:
```diff
@@ -2,6 +2,7 @@ import * as Utils from "./common/utils";
 import Renderer = require("./renderer");
 import Ruler = require("./ruler");
 
+
```
Extraneous whitespace change
